### ISSUE TITLE
Fix `function-url-quotes` false positives for SCSS variable and `#` character 

### ISSUE THREAD

**Issue by vimalloc** [stylelint/stylelint#7851]
### What minimal example or steps are needed to reproduce the bug?

```scss 
url($src-prefix + '.eot?#iefix')
```

### What minimal configuration is needed to reproduce the bug?

```yaml
extends:
  - stylelint-config-standard-scss
```

### How did you run Stylelint?

`./node_modules/stylelint/bin/stylelint.mjs 'app/assets/stylesheets/**/*' --fix`

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "^16.6.1",
  "stylelint-config-standard-scss": "^13.1.0"
}
```

### What did you expect to happen?

nothing should happen, the format is valid as is:

```scss 
url($src-prefix + '.eot?#iefix')
```

### What actually happened?

It tries to double quote everything passed into `url`:

```scss
url("$src-prefix + '.eot?#iefix'")
```

### Do you have a proposal to fix the bug?

Probably related to #7404 and #7416

**Comment by ybiquitous** [comment]
@vimalloc Thanks for the report with reproducible info. I can reproduce the behavior with [this demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLgIyTAGsBzAJwgFdMp4VKyAbACgBIBnMsAWgAcyYAMwCWADxQBqFAHIAdDAg4A-AGJhQsdICUAbiwBfEABoQIxjABySALZxEMUTd7nZ7MO3bHwETCJIIQDEwUFHQQBxwYGnYwugBtLBCQsPYcAE9zRmFMHG5IX2ESblSkGiQyKGL3GJBElABdAy98vwAxCDJrJBwAgCt2Hy9YXk9EIOSQVIyYLJzY0JBGbphUsKM6lPTM7NyWwuKcUqhyyrcPebClyNXazEN9IA). 👍🏼 

As you guessed, it seems necessary to fix `isStandardSyntaxUrl()` to ignore the reported pattern:
https://github.com/stylelint/stylelint/blob/67c21375e12f695a3929bde27281d772dcdeaf93/lib/utils/isStandardSyntaxUrl.mjs#L15

I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by vimalloc** [comment]
Sure, I'll try to get a PR up sometime this week. Thanks!
